edb/pgsql/delta.py

```diff
@@ -2252,36 +2280,337 @@ def adjust_pointer_storage(self, pointer, schema, orig_schema, context):
                 self.pgops.add(opg)
 
             else:
-                if old_target != new_target and not type_change_ok:
-                    if not isinstance(old_target, s_objtypes.ObjectType):
-                        source = source_ctx.scls
-
-                        self.pgops.add(self.drop_inhview(
-                            schema,
-                            context,
-                            source,
-                            drop_ancestors=True,
-                        ))
+                if (
+                    old_target != new_target
+                    and not type_change_ok
+                    and not pointer.is_endpoint_pointer(orig_schema)
+                ):
+                    self._alter_pointer_type(
+                        pointer, schema, orig_schema, context)
+
+    def _alter_pointer_type(self, pointer, schema, orig_schema, context):
```

Comment:
This function is a doozy; it is probably worth considering whether it can be split up in a way that makes it easier to read/think about. (Maybe it can't be, though)

Reply:
I tried, but the result looked even harder to follow for me and was much longer.